### DISCUSSION TITLE
Option in the Article Type Section to add a URL to server of choice?

### DISCUSSION THREAD

**Discussion by paurray** [templaza/astroid-framework#716] [Ideas]
Hello Astroid Community,

Would it be worth considering adding an option in the Article Type Section to add a URL to server of choice eg local server?
Rather than having the choice between either You Tube or Vimeo?

just a thought

Paul

<img width="960" alt="CustomURL" src="https://github.com/templaza/astroid-framework/assets/9163424/63db55ac-e8f3-4ba5-b93d-ff3d3259b2ff">

**Comment by Chacapamac** [comment]
That seem to be a good idea @paurray . I second.


———————————————————————————————————————
 To Everybody, Please let’s “**Buy a Coffee**” for the Great Work done on Astroid.
 **I participate —>** https://ko-fi.com/astroidframework/goal?g=0
———————————————————————————————————————

**Comment by sonvnn** [comment]
Thank you @paurray. I will note it in 3.1.1.

**Comment by paurray** [comment]
@sonvnn @Chacapamac that would be awesome if you could do that I think this is a good example of how something like this could work: https://www.akeeba.com/videos/1212-akeeba-backup.html

Very busy but here hope to add another couple of videos here (Astroid Boilerplate Template Deep Dive) and then a new Blog Features playlist by end of 2024!
https://www.youtube.com/playlist?list=PLOd7DxRu2yaIgxI86tWdSfmxGUI19Kczu

**Comment by Chacapamac** [comment]
Are these videos somehow link to the official Astroid support? I do not see anything?

**Comment by paurray** [comment]
@Chacapamac at the end of each video description you can see the following if I missed this some where in one of the 16 video descriptions am happy to add if you let me know :-)

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
Get Astroid Help  ➡️ https://github.com/templaza/astroid-framework/discussions
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

**Reply by Chacapamac** [reply]
I mean, it would be great if the Astroid website had a link to these videos or, even better if embedded in the Astroid YouTube Channel as a playlist.

**Reply by paurray** [reply]
@Chacapamac
In case you missed it. Currently the Astroid Website has a sub menu “Hands On” in the Blog Section which is basically a Blog Category called Hands On.”

I think there is room for improvement with maybe better naming on my part:

- eg tutorial: 201 Blah, 202 Blah etc.
- More subcategories
- and or Tags.

Maybe this can be fine tuned moving forward. What I do feel is mission critical is to have all the hand on tutorials in one place as is kind of the situation now.

<img width="960" alt="AstroidBlogCategory" src="https://github.com/user-attachments/assets/fd170ff1-7d68-4ec3-b3f1-2a45f5e2f4c0">


https://astroidframe.work/blog/astroid-hands-on

FYI I just have very basic author control. I can add and edit articles and think I can even add tags. That is about it. No acces to offical Astroid YT Channel. This kind of level of control is fine with me :-)

With respect to the official Astroid You Channel we discussed this before. 

- I have the official Astroid You Tube Channel in the “Other Useful Resources” section on my unofficial Astroid Channel. Ditto Joomplates.

- Would be happy to be linked in a similar way from official Astroid or for that matter Joomplates channel.

- Also pretty sure that there is a way to add playlists from other channels to other channels. No time to watch now but maybe this helps: https://www.youtube.com/watch?v=9xQ3YrYqEfc&t=2s

<img width="960" alt="MyChannel" src="https://github.com/user-attachments/assets/4e91a772-7ac7-47fc-947f-c39e55bde221">

**Comment by rinka88** [comment]
Initially, the server should be optimal for video output, and when the site is under load, when there are a lot of users watching

**Reply by paurray** [reply]
@rinka88 
My use case (Short Term) is to have a couple of dozen videos that can only be viewed by people (logged in) who I chose to give access to. eg Samples of my work that I need to host myself and control access to for legal reasons . Views are like a couple a month = extremely low key. Less is more!

…Long Term I think it could be interesting for training purposes. This is a recurring area of interest for me but no immediate plans to go down this route currently. Just had another little peak down the rabbit hole and that was enough for the moment eg here:

https://www.reddit.com/r/webdev/comments/2gz5vy/how_difficult_is_it_to_host_video_on_your_own/

But maybe there are others who have other Use Cases for functionality like this?

At the end of the day I just think that Astroid with its extended Blog functionality combined with a Bare Bones sensible use of Joomla Categories and more video options (eg choosing where to embed them) would be a really powerful combination.

In a weird round about way this article here and in particular the “videos” example in the “Override instead of adding cruft section” really made me bite the Astroid bullet! I could be wrong but I don't think there is much need to override with Astroid and all its countless options. Hope this makes some kind of sense.

https://www.dionysopoulos.me/joomla-performance-tuning-i-start-at-the-beginning.html

**Comment by Chacapamac** [comment]
> At the end of the day I just think that Astroid with its extended Blog functionality combined with a Bare Bones sensible use of Joomla Categories and more video options (eg choosing where to embed them) would be a really powerful combination.

I LIKE THAT!

**Comment by paurray** [comment]
I really can not express how this update has exceed my wildest dreams. It is way better than what I was thinking of. The way it is implemented is perfect. Being able to add multiple videos with or with out auto play etc etc :-)

I am off for a holiday for most of August and maybe my brain is just fried at this point. But I can not help wondering if it maybe makes sense to also add this new video functionality to the “Article Type Section” as shown in the first screenshot at the start of this thread?

My reasoning

1) When sorting articles in a category view eg like here:

https://astroidframe.work/blog/astroid-hands-on

If videos were coming from different sources eg YT. Vimeo or Local this would be immediately apparent. Obviously this would also require a custom Icon for a local blog posts containing video hosted locally displayed in a listing. 

<img width="960" alt="001-Category-VIEW" src="https://github.com/user-attachments/assets/575e4167-05b1-4c3d-b127-ebd933f1e40c">

2) I have this nagging feeling that it just feels more holistic to have pretty much the same kind of logic/options available in different places! As in same workflow in Widget as in Article Options. This way people can chose what works best for them. And avoid possible  confusion.

3) No idea if this makes any sense but if inserting video using the Tabs in the Joomla Blog Article section the Open graph options are just a couple of Tabs away.

<img width="960" alt="002-Open-graph-Options" src="https://github.com/user-attachments/assets/7a39c14e-4966-4bcf-b355-d7b71eb17823">

Hope this makes sense and hope I am not being too fussy. Back at end of August with recharged batteries and brain! One way or the other Thank you :-) from an over the moon Astroid Camper.

**Comment by sonvnn** [comment]
@paurray I added this feature to 3.1.2. You can check it here https://github.com/templaza/astroid-framework/commit/bf35c8fd0bf04d341c31ac15bf613e5053b749aa

**Comment by paurray** [comment]
@sonvnn 
thank you for yet another awesome addition to Astroid!
